apps/admin/backend/src/cast_vote_records.ts

```diff
@@ -81,6 +100,79 @@ function validateCastVoteRecordAgainstElectionDefinition(
   return ok();
 }
 
+/**
+ * Lists the cast vote record exports on the inserted USB drive
+ */
+export async function listCastVoteRecordExportsOnUsbDrive(
+  usbDrive: UsbDrive,
+  electionDefinition: ElectionDefinition
+): Promise<
+  Result<CastVoteRecordFileMetadata[], 'no-usb-drive' | 'permission-denied'>
+> {
+  const { election, electionHash } = electionDefinition;
+
+  const listDirectoryResult = await listDirectoryOnUsbDrive(
+    path.join(
+      SCANNER_RESULTS_FOLDER,
+      generateElectionBasedSubfolderName(election, electionHash)
+    ),
+    buildGetUsbDrivesFn(usbDrive)
+  );
+  if (listDirectoryResult.isErr()) {
+    const errorType = listDirectoryResult.err().type;
+    switch (errorType) {
+      case 'no-entity':
+      case 'not-directory': {
+        return ok([]);
+      }
+      case 'no-usb-drive':
+      case 'usb-drive-not-mounted': {
+        return err('no-usb-drive');
+      }
+      case 'permission-denied': {
+        return err('permission-denied');
+      }
+      default: {
+        throwIllegalValue(errorType);
+      }
+    }
+  }
+
+  const castVoteRecordExportSummaries: CastVoteRecordFileMetadata[] = [];
+  for (const entry of listDirectoryResult.ok()) {
+    if (entry.type === FileSystemEntryType.Directory) {
+      const exportDirectoryNameComponents =
+        parseCastVoteRecordReportExportDirectoryName(entry.name);
+      if (!exportDirectoryNameComponents) {
+        continue;
+      }
+      const metadataResult = await readCastVoteRecordExportMetadata(entry.path);
+      if (metadataResult.isErr()) {
+        continue;
+      }
```

Comment:
This could happen if a directory appears to be a valid export directory, by name, but actually isn't and doesn't have a `metadata.json` file. Or has a malformed `metadata.json`, say, because someone modified it. In practice, these should be exceedingly rare, unless someone has manually modified the contents of a scanner USB (which users know not to do). That's why I think it's fine to just ignore these directories instead of failing loudly.